worker/task/process_broadcast_event_creation.go

```diff
@@ -0,0 +1,92 @@
+package task
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/frain-dev/convoy/util"
+
+	"github.com/frain-dev/convoy/pkg/msgpack"
+
+	"github.com/frain-dev/convoy/api/models"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/pkg/log"
+	"github.com/frain-dev/convoy/queue"
+	"github.com/hibiken/asynq"
+	"github.com/oklog/ulid/v2"
+)
+
+func ProcessBroadcastEventCreation(endpointRepo datastore.EndpointRepository, eventRepo datastore.EventRepository, projectRepo datastore.ProjectRepository, eventDeliveryRepo datastore.EventDeliveryRepository, eventQueue queue.Queuer, subRepo datastore.SubscriptionRepository, deviceRepo datastore.DeviceRepository) func(context.Context, *asynq.Task) error {
+	return func(ctx context.Context, t *asynq.Task) error {
+		var broadcastEvent models.BroadcastEvent
+
+		err := msgpack.DecodeMsgPack(t.Payload(), &broadcastEvent)
+		if err != nil {
+			err := json.Unmarshal(t.Payload(), &broadcastEvent)
+			if err != nil {
+				return &EndpointError{Err: err, delay: defaultDelay}
+			}
+		}
+
+		project, err := projectRepo.FetchProjectByID(ctx, broadcastEvent.ProjectID)
+		if err != nil {
+			return &EndpointError{Err: err, delay: 10 * time.Second}
+		}
+
+		var isDuplicate bool
+		if len(broadcastEvent.IdempotencyKey) > 0 {
+			events, err := eventRepo.FindEventsByIdempotencyKey(ctx, broadcastEvent.ProjectID, broadcastEvent.IdempotencyKey)
+			if err != nil {
+				return &EndpointError{Err: err, delay: 10 * time.Second}
+			}
+
+			isDuplicate = len(events) > 0
+		}
+
+		subscriptions, err := subRepo.FindSubscriptionByEventType(ctx, project.UID, broadcastEvent.EventType)
```

Comment:
@danvixent We should retrieve all subscriptions in the project and run both `matchSubscriptions` and `matchSubscriptionsUsingFilter` routines in `process_event_creation.go` against them.